server/scripts/parse-xml.js

```diff
@@ -0,0 +1,125 @@
+const fs = require('fs');
+const path = require('path');
+const axios = require('axios');
+const inquirer = require('inquirer');
+const parser = require('fast-xml-parser');
+const _ = require('lodash');
+const asyncPool = require('tiny-async-pool');
+const { validate, FormSchema } = require('../validation');
+
+const endpoints = [
+  { name: 'Local', value: 'http://localhost:4000' },
+  { name: 'Dev', value: 'https://hcap-server-rupaog-dev.pathfinder.gov.bc.ca' },
+];
+
+const postHcapSubmission = async (endpoint, data) => {
+  const apiUrl = `${endpoint}/api/v1`;
+  const response = await axios.post(`${apiUrl}/form`, data);
+  return response.data;
+};
+
+const getDirContent = (pathString) => {
+  try {
+    const fileNames = fs.readdirSync(pathString).filter((file) => (file.slice(-4) === '.xml'));
+    const filesString = [];
+
+    fileNames.forEach((name) => {
+      const filePath = path.join(pathString, name);
+      filesString.push({ name, content: fs.readFileSync(filePath).toString() });
+    });
+
+    return filesString;
+  } catch (e) {
+    if (typeof pathString === 'undefined') throw Error('Provide a path to a folder containing XMLs as the first command line argument');
+    throw Error(`Could not read contents of file ${pathString}`);
+  }
+};
+
+const fromXmlStrings = (xmlStrings) => parser.parse(xmlStrings);
+
+const getUserInput = async () => { // Prompt user for query, output format
+  const { 'Select Endpoint': endpoint } = await inquirer.prompt([{ name: 'Select Endpoint', type: 'list', choices: endpoints }]);
+  const { Proceed: proceed } = await inquirer.prompt([{ name: 'Proceed', type: 'confirm' }]);
+  if (!proceed) throw Error('User cancelled operation');
+  return { endpoint };
+};
+
+const extractPreferredLocations = (preferredLocationString) => {
+  const preferredLocations = preferredLocationString.split(' ');
+  return preferredLocations.map((item) => {
+    switch (item) {
+      case 'Coastal':
+        return 'Vancouver Coastal';
+      case 'Island':
+        return 'Vancouver Island';
+      default:
+        return item;
+    }
+  });
+};
+
+const extractOrbeonId = (name) => name.match(/^Health Career Access Program - Expression of Interest - ([\w-]{16})( \([0-9]+\))?.xml$/)[1];
```

Comment:
Discovered this falls over pretty hard if you have issues in your XML file names, e.g. if you're missing the space between the - and the ID.

```
 File error: Health Career Access Program - Expression of Interest -hcc9tlb0srg8bned.xml
TypeError: Cannot read property '1' of null
```

I'm wondering if it makes the most sense just to include the expected file name pattern in getDirContent? Maybe it would make sense for the script to exit there if it finds any files that are not XML, or any files that don't match the expected XML filename format?

Reply:
Nice catch. I added the file name validation to the error array, this way it will be part of all validations.